pkg/authorization/registry/clusterrole/proxy.go

```diff
@@ -0,0 +1,165 @@
+package clusterrole
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metainternal "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	restclient "k8s.io/client-go/rest"
+	rbacinternalversion "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
+	"github.com/openshift/origin/pkg/authorization/registry/util"
+)
+
+type REST struct {
+	client   restclient.Interface
```

Comment:
> ok to call it impersonatingClient ?

I wouldn't know if that was the client to impersonate with or the one doing the impersonation.  "privileged" is used throughout much of the codebase.